### PR TITLE
*: fix issue of multi-schema change with foreign key

### LINKED ISSUE #40037: issue of foreign key with multi-schema change
## reproduce

```
create table t1 (id int key);
create table t2 (a int, b int);
alter table t2 add foreign key (a) references t1(id), add foreign key (b) references t1(id);  --failed, but should success.
(8200, "Unsupported operate same index 'fk_1'")
```

version: master,  0c1808245b4f5960266c7f4ee860408a36b58552

### PR DESCRIPTION
**Pull request by crazycs520**
Signed-off-by: crazycs520 <crazycs520@gmail.com>

### What problem does this PR solve?

Issue Number: close #40037

Problem Summary: fix issue of multi-schema change with foreign key.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: ddl/fktest/foreign_key_test.go [hunk 1]
```diff
@@ -1544,3 +1544,33 @@ func getLatestSchemaDiff(t *testing.T, tk *testkit.TestKit) *model.SchemaDiff {
...
+	tk.MustExec("set @@foreign_key_checks=1;")
+	tk.MustExec("use test")
+	tk.MustExec("create table t1 (id int key);")
+	tk.MustExec("create table t2 (a int, b int);")
+	tk.MustExec("alter table t2 add foreign key (a) references t1(id), add foreign key (b) references t1(id)")
+	tk.MustExec("alter table t2 add column c int, add column d int")
+	tk.MustExec("alter table t2 add foreign key (c) references t1(id), add foreign key (d) references t1(id), add index(c), add index(d)")
+	tk.MustExec("drop table t2")
+	tk.MustExec("create table t2 (a int, b int, index idx1(a), index idx2(b));")
+	err := tk.ExecToErr("alter table t2 drop index idx1, drop index idx2, add foreign key (a) references t1(id), add foreign key (b) references t1(id)")
```
Comment (hawkingrei) [inline ddl/fktest/foreign_key_test.go:1560]: tk.MustGetErrMsg will be better.

**Review by hawkingrei — APPROVED** [review]

File: ddl/multi_schema_change.go [hunk 2]
```diff
@@ -323,6 +326,41 @@ func checkOperateSameColAndIdx(info *model.MultiSchemaInfo) error {
...
+}
+
+func findIndexByColumns(tbInfo *model.TableInfo, indexes []*model.IndexInfo, cols ...model.CIStr) *model.IndexInfo {
+	for _, index := range indexes {
+		if model.IsIndexPrefixCovered(tbInfo, index, cols...) {
+			return index
+		}
+	}
+	return nil
+}
```
Comment (tangenta) [inline ddl/multi_schema_change.go:362]: Could you move this to `parser/model/model.go` and update `model.FindIndexByColumns`?

Comment (crazycs520) [reply ddl/multi_schema_change.go]: done.

**Review by tangenta — APPROVED** [review]

**Pull request merged**